assets/src/design-system/components/contextMenu/stories/index.js

```diff
@@ -293,3 +292,272 @@ export const QuickActionMenu = () => {
+        'web-stories'
+      ),
+    },
+  },
+  {
+    label: __('Paste style', 'web-stories'),
+    shortcut: {
+      display: '⌥⌘V',
+      title: _x(
+        'Option Command V',
+        'The keyboard keys "Option" "Command" and the letter "V"',
+        'web-stories'
+      ),
+    },
+  },
+  { label: __('Clear style', 'web-stories') },
+];
+
+const pageElement = [
+  ...rightClickMenuMainOptions,
+  ...rightClickMenuPageAddOptions,
+  ...rightClickMenuPageDeleteOptions,
+];
+
+const shapeElement = [
+  ...rightClickMenuMainOptions,
+  ...rightClickMenuLayeringOptions,
+  ...rightClickMenuStyleOptions,
+  { label: __('Add color to "Saved colors"', 'web-stories') },
+];
+
+const foregroundMediaElement = [
+  ...rightClickMenuMainOptions,
+  ...rightClickMenuLayeringOptions,
+  { label: __('Set as page background', 'web-stories'), separator: 'top' },
+  { label: __('Scale & crop image', 'web-stories') },
+  ...rightClickMenuStyleOptions,
+];
+
+const backgroundMediaElement = [
+  ...rightClickMenuMainOptions,
+  {
+    label: __('Detach image from background', 'web-stories'),
+    separator: 'top',
+  },
+  { label: __('Replace background image', 'web-stories') },
+  { label: __('Scale & crop background image', 'web-stories') },
+  { label: __('Clear style', 'web-stories') },
+  ...rightClickMenuPageAddOptions,
+  ...rightClickMenuPageDeleteOptions,
+];
+
+const textElement = [
+  ...rightClickMenuMainOptions,
+  ...rightClickMenuLayeringOptions,
+  ...rightClickMenuStyleOptions,
+  { label: __('Add style to "Saved styles"', 'web-stories') },
+  { label: __('Add color to "Saved colors"', 'web-stories') },
+];
+
+const SampleLayout = styled.div`
+  display: block;
+  width: 400px;
+  height: 800px;
+  border: 1px solid black;
+`;
+
+const RightClickContextMenuContainer = styled.div`
+  position: absolute;
+  top: ${({ position }) => position?.y ?? 0}px;
+  left: ${({ position }) => position?.x ?? 0}px;
+`;
+
+export const RightClickMenu = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [menuPosition, setMenuPosition] = useState({});
+  const layoutRef = useRef();
+
+  const generateMenuItemsWithEventHandler = (i) =>
+    i.map((item) => ({
+      ...item,
+      onClick: () => {
+        action(`Clicked on \`${item.label}\``)();
+        setIsOpen(false);
+      },
+    }));
+  const handleMenu = (e) => {
+    e.preventDefault();
+    const layoutRect = layoutRef?.current?.getBoundingClientRect();
+
+    setIsOpen(true);
+    setMenuPosition({
+      x: e.clientX - layoutRect?.left,
+      y: e.clientY - layoutRect?.top,
+    });
+  };
+  return (
+    <ViewportContainer>
+      {/*eslint-disable-next-line styled-components-a11y/no-noninteractive-element-interactions*/}
+      <SampleLayout
+        ref={layoutRef}
+        onClick={handleMenu}
+        // TODO: confirm we don't need this menu to  show up for keyboards since they have a separate menu
+        role="region"
```

Comment:
layout in canvas is this region, that's why i grabbed it. added onKeyDown to stop the linter yelling at me.